cartridge-cli/utils.lua

```diff
@@ -0,0 +1,450 @@
+    local searchlist = {...}
+    local estring,evar
+
+    local statements = {'do', 'if', 'for', 'while', 'repeat'}
+
+    function estring(str)
+        local b,e,i
+        b,i = string.find(str, '%$.')
+        if not b then return str end
+
+        local R, pos = {}, 1
+        repeat
+            b,e = string.find(str, '^%b{}', i)
+            if b then
+                table.insert(R, string.sub(str, pos, b-2))
+                table.insert(R, evar(string.sub(str, b+1, e-1)))
+                i = e+1
+                pos = i
+            else
+                b,e = string.find(str, '^%b()', i)
+                if b then
+                    table.insert(R, string.sub(str, pos, b-2))
+                    table.insert(R, evar(string.sub(str, b+1, e-1)))
+                    i = e+1
+                    pos = i
+                elseif string.find(str, '^%a', i) then
+                    table.insert(R, string.sub(str, pos, i-2))
+                    table.insert(R, evar(string.sub(str, i, i)))
+                    i = i+1
+                    pos = i
+                elseif string.find(str, '^%$', i) then
+                    table.insert(R, string.sub(str, pos, i))
+                    i = i+1
+                    pos = i
+                end
+            end
+            b,i = string.find(str, '%$.', i)
+        until not b
+
+        table.insert(R, string.sub(str, pos))
+        return table.concat(R)
+    end
+
+    local function search(index)
+        for _,symt in ipairs(searchlist) do
+            local ts = type(symt)
+            local value
+            if     ts == 'function' then value = symt(index)
+            elseif ts == 'table'
+            or ts == 'userdata' then value = symt[index]
+                if type(value)=='function' then value = value(symt) end
+            else error'search item must be a function, table or userdata' end
+            if value ~= nil then return value end
+        end
+        error('unknown variable: '.. index)
+    end
+
+    local function elist(var, v, str, sep)
+        local tab = search(v)
+        if tab then
+            assert(type(tab)=='table', 'expecting table from: '.. var)
+            local R = {}
+            table.insert(searchlist, 1, tab)
+            table.insert(searchlist, 1, false)
+            for _,elem in ipairs(tab) do
+                searchlist[1] = elem
+                table.insert(R, estring(str))
+            end
+            table.remove(searchlist, 1)
+            table.remove(searchlist, 1)
+            return table.concat(R, sep)
+        else
+            return ''
+        end
+    end
+
+    local function get(tab,index)
+        for _,symt in ipairs(searchlist) do
+            local ts = type(symt)
+            local value
+            if     ts == 'function' then value = symt(index)
+            elseif ts == 'table'
+            or ts == 'userdata' then value = symt[index]
+            else error'search item must be a function, table or userdata' end
```

Comment:
I think that this part of code should be rewritten using our styleguide

Reply:
This function was copy-pasted from some resource (there is a comment and a link).
Maybe it should be "as is".

Anyway, it requires many edits, it can be performed in the next PR (for example, the one that will continue restructuring `cartridge-cli`).

Reply:
Ok. Agree